images/metering-ansible-operator/roles/meteringconfig/tasks/configure_metastore.yml

```diff
@@ -0,0 +1,43 @@
+- name: Get metastore secret
+  block:
+    - name: Get Metastore storage credentials secret
+      k8s_info:
+        api_version: v1
+        kind: Secret
+        name: "{{ meteringconfig_spec_overrides.hive.spec.config.db.secretName }}"
+        namespace: "{{ meta.namespace }}"
+      register: metastore_secret
+  when: meteringconfig_spec_overrides.hive.spec.config.db.secretName is defined and meteringconfig_spec_overrides.hive.spec.config.db.secretName | length > 0
+
+- name: Create variables for the decoded username and password data
+  set_fact:
+    _username: "{{ metastore_secret.resources[0].data.username | b64decode }}"
+    _password: "{{ metastore_secret.resources[0].data.password | b64decode }}"
+  no_log: true
+
+- name: Create a dictionary of these decoded values
+  set_fact:
+    _metastore_overrides:
```

Comment:
We should move this dictionary to defaults/main.yml now that we have a simple POC working.